homeassistant/components/synology_dsm/camera.py

```diff
@@ -94,7 +98,27 @@ def camera_image(self) -> bytes:
         """Return bytes of camera image."""
         if not self.available:
             return None
-        return self._api.surveillance_station.get_camera_image(self._camera.id)
+        try:
+            return self._api.surveillance_station.get_camera_image(self._camera.id)
+        except (SynologyDSMAPIErrorException) as err:
+            _LOGGER.debug(
+                "SynoDSMCamera.camera_image() - Exception:%s",
+                err,
+            )
+            return None
+
+    @property
+    def should_poll(self) -> bool:
+        """Return the polling state."""
+        return True
+
+    async def async_update(self):
+        """Update camera state."""
+        await self.hass.async_add_executor_job(self._api.surveillance_station.update)
+        await self.hass.async_block_till_done()
```

Comment:
I assume this can wait on work scheduled by other integrations.  Is it enough to just await on the result of async_add_executor_job this like the next line does?

Reply:
Hi @allenporter 
sorry, but I´m not sure if I get it correct, what you mean?

Reply:
What is the intent of this `await self.hass.async_block_till_done()` call?

I often see async_block_until_done used in tests so that the test can lets all the work scheduled by hass complete before asserting on state.  Typically in a test, this only includes test specific tasks.

Here, I imagine home assistant could be quite busy and getting lots of different different tasks assigned from various integrations and work.  That is, this async_block_till_done() call won't just be waiting on this executor job to complete, but everything home assistant is doing.

I assume the intent is to wait until the update is complete.  I was assuming that await on the return value of async_add_executor_job is sufficient, following the example from 
https://developers.home-assistant.io/docs/asyncio_working_with_async#calling-sync-functions-from-async or the line of code on 119

Reply:
This isn't needed if the sync API is used.